Jenkinsfile

```diff
@@ -28,292 +28,305 @@
                         }
                     }
                 }
             }
-            dir('Palisade-examples') {
-                git branch: 'develop', url: 'https://github.com/gchq/Palisade-examples.git'
-                if (sh(script: "git checkout ${GIT_BRANCH_NAME}", returnStatus: true) == 0) {
+
+            stage('Integration Tests') {
+                // Always run some sort of integration test
+                // If this branch name exists in integration-tests, use that
+                // Otherwise, default to integration-tests/develop
+                dir('Palisade-integration-tests') {
+                    git branch: 'develop', url: 'https://github.com/gchq/Palisade-integration-tests.git'
+                    if (sh(script: "git checkout ${GIT_BRANCH_NAME}", returnStatus: true) == 0) {
+                        INTEGRATION_REVISION = "${GIT_BRANCH_NAME_LOWER}-BRANCH-SNAPSHOT"
+                    }
                     container('docker-cmds') {
                         configFileProvider([configFile(fileId: "${env.CONFIG_FILE}", variable: 'MAVEN_SETTINGS')]) {
-                            sh 'mvn -s $MAVEN_SETTINGS install -P quick -pl hr-data-generator/'
+                            sh "mvn -s ${MAVEN_SETTINGS} -D revision=${INTEGRATION_REVISION} -D common.revision=${COMMON_REVISION} -D services.revision=${SERVICES_REVISION} -D examples.revision=${EXAMPLES_REVISION} install"
                         }
                     }
                 }
             }
-        }
 
-        stage('Install, Unit Tests, Checkstyle') {
-            dir('Palisade-services') {
-                git branch: GIT_BRANCH_NAME, url: 'https://github.com/gchq/Palisade-services.git'
-                container('docker-cmds') {
-                    configFileProvider([configFile(fileId: "${env.CONFIG_FILE}", variable: 'MAVEN_SETTINGS')]) {
-                        sh 'mvn -s $MAVEN_SETTINGS install'
+            stage('SonarQube Analysis') {
+                dir('Palisade-services') {
+                    container('docker-cmds') {
+                        withCredentials([string(credentialsId: "${env.SQ_WEB_HOOK}", variable: 'SONARQUBE_WEBHOOK'),
+                                         string(credentialsId: "${env.SQ_KEY_STORE_PASS}", variable: 'KEYSTORE_PASS'),
+                                         file(credentialsId: "${env.SQ_KEY_STORE}", variable: 'KEYSTORE')]) {
+                            configFileProvider([configFile(fileId: "${env.CONFIG_FILE}", variable: 'MAVEN_SETTINGS')]) {
+                                withSonarQubeEnv(installationName: 'sonar') {
+                                        sh "mvn -s ${MAVEN_SETTINGS} org.sonarsource.scanner.maven:sonar-maven-plugin:3.7.0.1746:sonar -Dsonar.projectKey=Palisade-Services-${GIT_BRANCH_NAME} -Dsonar.projectName=Palisade-Services-${GIT_BRANCH_NAME} -Dsonar.webhooks.project=$SONARQUBE_WEBHOOK -Djavax.net.ssl.trustStore=$KEYSTORE -Djavax.net.ssl.trustStorePassword=$KEYSTORE_PASS"
```

Comment:
this doesnt work and just creates a project key called Palisade-Services-
revert back to the bulky if statement